# CENTRAL TENDENCY - PROJECT

## Dataset Description

###  Swing_States_Dataset

2008_Swing_States dataset (Available on Kaggle) contains US swing states election results. 

There are 221 observations in the dataset with 6 features each (state, county, total_votes, dem_votes, rep_votes, dem_share). 

There are no null values in the dataset.

# Problem

**1. What is the highest dem_share percentage for different states? Find the countys with highest dem_share in each state? Also check for null values.**

**2. Whats the average total_votes for each states? Also calculate their respective median.**

**3. Calculate the variablity and skewness (whether positive or negative or zero) for total_votes for each states.**

**4. Also calculate the z - score of total_votes for each states.**

**5. Calculate Interquartile range of total_votes of state PA.**

# Solution

# 1. 

In [1]:
import numpy as np
import pandas as pd

In [2]:
df=pd.read_csv("2008_swing_states.csv")

In [3]:
df.head()

state           county  total_votes  dem_votes  rep_votes  dem_share
0    PA      Erie County       127691      75775      50351      60.08
1    PA  Bradford County        25787      10306      15057      40.64
2    PA     Tioga County        17984       6390      11326      36.07
3    PA    McKean County        15947       6465       9224      41.21
4    PA    Potter County         7507       2300       5109      31.04

In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 222 entries, 0 to 221
Data columns (total 6 columns):
state          222 non-null object
county         222 non-null object
total_votes    222 non-null int64
dem_votes      222 non-null int64
rep_votes      222 non-null int64
dem_share      222 non-null float64
dtypes: float64(1), int64(3), object(2)
memory usage: 10.5+ KB


In [51]:
df.isnull().sum()

state          0
county         0
total_votes    0
dem_votes      0
rep_votes      0
dem_share      0
dtype: int64

In [5]:
df['state'].unique()

array(['PA', 'OH', 'FL'], dtype=object)

In [6]:
df.columns[1]

'county'

In [8]:
state = df.state
type(state)

pandas.core.series.Series

In [9]:
state

0      PA
1      PA
2      PA
3      PA
4      PA
5      PA
6      PA
7      PA
8      OH
9      OH
10     PA
11     OH
12     OH
13     OH
14     OH
15     PA
16     PA
17     PA
18     PA
19     PA
20     OH
21     OH
22     PA
23     PA
24     PA
25     PA
26     OH
27     OH
28     PA
29     OH
       ..
192    FL
193    FL
194    FL
195    FL
196    FL
197    FL
198    FL
199    FL
200    FL
201    FL
202    FL
203    FL
204    FL
205    FL
206    FL
207    FL
208    FL
209    OH
210    OH
211    OH
212    OH
213    OH
214    OH
215    OH
216    OH
217    OH
218    OH
219    OH
220    OH
221    OH
Name: state, Length: 222, dtype: object

In [56]:
state[0]

'PA'

In [57]:
state[1:3]

1    PA
2    PA
Name: state, dtype: object

In [11]:
df.state.value_counts().sort_index()

FL    67
OH    88
PA    67
Name: state, dtype: int64

In [12]:
a=df[df['state']=='PA']
b=df[df['state']=='OH']
c=df[df['state']=='FL']

In [13]:
a.head()

state           county  total_votes  dem_votes  rep_votes  dem_share
0    PA      Erie County       127691      75775      50351      60.08
1    PA  Bradford County        25787      10306      15057      40.64
2    PA     Tioga County        17984       6390      11326      36.07
3    PA    McKean County        15947       6465       9224      41.21
4    PA    Potter County         7507       2300       5109      31.04

In [14]:
b.head()

state            county  total_votes  dem_votes  rep_votes  dem_share
8     OH  Ashtabula County        44874      25027      18949      56.94
9     OH       Lake County       121335      60155      59142      50.46
11    OH      Lucas County       219830     142852      73706      65.99
12    OH     Fulton County        21973       9900      11689      45.88
13    OH     Geauga County        51102      21250      29096      42.23

In [15]:
c.head()

state             county  total_votes  dem_votes  rep_votes  dem_share
142    FL     Jackson County        21565       7671      13717      35.86
143    FL    Escambia County       154447      61572      91411      40.25
144    FL  Santa Rosa County        76185      19470      55972      25.81
145    FL    Okaloosa County        95529      25872      68789      27.33
146    FL      Holmes County         8589       1446       7033      17.06

In [16]:
a['dem_share'].max()

83.56

In [17]:
a[a['dem_share']==83.56]

state               county  total_votes  dem_votes  rep_votes  dem_share
114    PA  Philadelphia County       717329     595980     117221      83.56

In [18]:
b['dem_share'].max()

69.64

In [19]:
b[b['dem_share']==69.64]

state           county  total_votes  dem_votes  rep_votes  dem_share
215    OH  Cuyahoga County       665352     458422     199880      69.64

In [67]:
c['dem_share'].max()

69.58

In [68]:
c[c['dem_share']==69.58]

state          county  total_votes  dem_votes  rep_votes  dem_share
150    FL  Gadsden County        22510      15582       6811      69.58

# Inference:

**The highest dem_share in PA state was for Philadelphia County. The share of votes for democratic party was 83.56%.**

**The highest dem_share in OH state was for Cuyahoga County. The share of votes for democratic party was 69.64%.**

**The highest dem_share in FL state was for Gadsden County. The share of votes for democratic party was 69.58%.**

# 2. 

In [69]:
a.mean()

total_votes    89479.656716
dem_votes      48900.940299
rep_votes      39640.074627
dem_share         45.476418
dtype: float64

In [70]:
b.mean()

total_votes    64640.465909
dem_votes      33217.431818
rep_votes      30397.829545
dem_share         44.318182
dtype: float64

In [71]:
c.mean()

total_votes    125234.985075
dem_votes       63911.552239
rep_votes       60382.447761
dem_share          41.325075
dtype: float64

In [72]:
a.median()

total_votes    37302.00
dem_votes      16780.00
rep_votes      20750.00
dem_share         44.03
dtype: float64

In [73]:
b.median()

total_votes    25687.500
dem_votes      11138.000
rep_votes      15418.500
dem_share         43.675
dtype: float64

In [74]:
c.median()

total_votes    49031.00
dem_votes      20907.00
rep_votes      30866.00
dem_share         41.85
dtype: float64

# Inference:

**The average value and median value of total votes for state PA is 89479 and 37302.<br>
The average value and median value of total votes for state OH is 89479 and 25687.<br>
The average value and median value of total votes for state FL is 125234 and 49031.**

# 3. 

In [75]:
a.var()

total_votes    1.814520e+10
dem_votes      8.569590e+09
rep_votes      2.290666e+09
dem_share      9.609935e+01
dtype: float64

In [76]:
b.var()

total_votes    1.168405e+10
dem_votes      4.575352e+09
rep_votes      1.690081e+09
dem_share      9.792486e+01
dtype: float64

In [77]:
c.var()

total_votes    3.212606e+10
dem_votes      1.132370e+10
rep_votes      5.532844e+09
dem_share      1.496768e+02
dtype: float64

In [78]:
a.skew()

total_votes    3.081416
dem_votes      4.118828
rep_votes      2.524580
dem_share      0.833819
dtype: float64

In [79]:
b.skew()

total_votes    3.782244
dem_votes      4.390079
rep_votes      3.027431
dem_share     -0.114473
dtype: float64

In [80]:
c.skew()

total_votes    2.319167
dem_votes      2.709455
rep_votes      1.866751
dem_share      0.182980
dtype: float64

# Inference:

**The variability and skewness of total votes for state PA is $1.18 * 10^{10}$ and 3.08.<br>
The variability and skewness of total votes for state OH is  $1.16 * 10^{10}$ and 3.78.<br>
The variability and skewness of total votes for state FL is $3.21 * 10^{10}$ and 2.31.**

# 4. 

In [81]:
from scipy import stats

In [82]:
stats.zscore(a['total_votes'])

array([ 0.28580965, -0.4764024 , -0.53476655, -0.55000272, -0.61313144,
       -0.49848249, -0.52502797, -0.53077987, -0.38405046, -0.57101321,
        0.13759925, -0.5625387 , -0.6510012 , -0.49495207, -0.65248966,
       -0.4876444 , -0.30100327, -0.64594491, -0.2667163 , -0.55864926,
       -0.53639713,  0.34178777, -0.52864815, -0.45937859, -0.41637027,
       -0.10259657, -0.15734803, -0.41542783, -0.60927191,  0.00707089,
       -0.44428454, -0.5391347 , -0.46790545, -0.3543635 ,  0.34486941,
       -0.19763373, -0.39027358, -0.55350322, -0.03733613, -0.54585148,
        0.47117174, -0.5299197 , -0.27209421, -0.17808925, -0.59664617,
        0.65367659,  0.67706563,  4.2032687 ,  0.29955735, -0.52159479,
        1.82089232, -0.23323713,  0.06408122,  2.49028868,  0.1781991 ,
       -0.50141454,  1.03711628, -0.19326558, -0.40623527,  1.23321175,
        0.78335226, -0.62211457, -0.27614821,  4.69612902, -0.33650196,
        1.55222125, -0.5497858 ])

In [83]:
stats.zscore(b['total_votes'])

array([-1.83913650e-01,  5.27504447e-01,  1.44393407e+00, -3.96992027e-01,
       -1.25966305e-01, -4.30264298e-01, -2.17827919e-01,  3.54991769e-03,
        7.64988073e-01,  3.93298844e-01, -4.63359787e-01, -4.22839446e-01,
        1.26218156e-01, -3.63412713e-01,  2.40149632e-01, -3.46041536e-01,
       -5.10542026e-01, -2.57352697e-01, -4.27631174e-01,  5.80511194e-01,
       -3.67264705e-01, -4.04435488e-01, -3.27369551e-02, -4.99302400e-01,
       -1.16289806e-01, -4.65109000e-01,  1.14354389e+00, -1.50297118e-01,
       -1.33772635e-01, -4.79419054e-01, -4.71612725e-01, -4.03523664e-01,
       -4.46584087e-01, -3.31452352e-01, -3.82914581e-01, -4.98037011e-01,
       -2.01815173e-01, -2.67047805e-01, -3.38765553e-01, -3.94721772e-01,
       -3.69497743e-01, -3.81221194e-01, -4.44537135e-01,  2.58432634e-01,
       -5.28983201e-01, -3.61449501e-01, -4.25705178e-01, -4.40238536e-01,
       -1.10102429e-01, -2.99873469e-01, -2.37906655e-01,  1.98138801e-02,
       -5.44009688e-01,  

In [84]:
stats.zscore(c['total_votes'])

array([-5.82760001e-01,  1.64209475e-01, -2.75724641e-01, -1.66986229e-01,
       -6.55701979e-01, -5.51949661e-01, -6.41412637e-01, -4.88665074e-01,
       -5.77447874e-01,  1.31386709e-01, -6.59254640e-01, -6.53914406e-01,
       -6.72577119e-01, -6.68883926e-01, -6.85556699e-01, -5.45867415e-01,
        1.63313365e+00, -6.41817370e-01, -2.47944180e-01, -6.04699932e-01,
       -6.51334230e-01, -6.23171521e-01, -6.85101374e-01, -1.09002528e-01,
       -6.63481857e-01, -1.72337706e-01, -6.38349029e-01, -6.74229781e-01,
       -6.70092505e-01,  1.59653286e-03, -6.60030380e-01, -5.17519203e-01,
       -6.63150201e-01, -4.28365398e-01, -5.98724491e-01,  2.06790817e-01,
        6.66624331e-01,  1.21931684e-01, -2.75876416e-01, -4.29281670e-01,
        4.53414075e-01,  9.14158241e-01,  1.89705364e+00, -2.09865499e-01,
        5.03842750e-01,  6.72296222e-01, -1.38087130e-01,  1.90026341e+00,
        2.18149700e+00, -3.07170188e-01, -4.52244677e-01, -6.62318248e-01,
        1.50420428e-01, -

# 5.

In [85]:
from scipy.stats import iqr

In [88]:
# Interquartile range
iqr(a.total_votes)

84711.0

# Inference:

**Interquartile range of total_votes of state PA is 84711.**